modules/wallet/signtransaction_test.go

```diff
@@ -0,0 +1,56 @@
+package wallet
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// mockHashSigner serves as a mock replacement for crypto.SignHash.
+type mockHashSigner struct {
+	mock.Mock
+}
+
+func (hs *mockHashSigner) SignHash(hash crypto.Hash, key crypto.SecretKey) (sig crypto.Signature, err error) {
```

Comment:
The mock serves two purposes:

1. Ensure that the dependency is being called in the expected fashion (e.g. called with correct parameters, called the correct number of times when that's relevant).
1. Ensure that the code behaves in expected fashion, given different results from the dependency (e.g. succeeds vs. fails).

Getting rid of the error parameter on `SignHash` means that (2) becomes unnecessary but (1) remains. For this case, I feel like (1) is the one we care about because otherwise it's difficult to test that `SignHash` is being called with the correct set of values to sign.